debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java

```diff
@@ -86,6 +90,23 @@ public final void start(Map<String, String> props) {
      */
     protected abstract ChangeEventSourceCoordinator start(Configuration config);
 
+    @Override
+    public final List<SourceRecord> poll() throws InterruptedException {
+        try {
+            return doPoll();
+        }
+        catch (RetriableException e) {
+            stop();
```

Comment:
Is this a robust solution? Let's suppose a temorary spike in postgres dropping the connectio. In this case the reconnect may fail (and I'd expect it to fail) and then we are back in the square one.

Reply:
Kafka Connect keeps retrying (for a configurable number of times, with back-off), so I'd expect it to work. At some point when connecting succeeds after retrying, things will be back to working state. What scenario are you envisioning specifically?

Reply:
@gunnarmorling Are you really sure that for tasks this is configurable with number and backoff? To me the connect code looks like - if the exception is `RetriableException` then retry indefinitely and as fast as possible - see `org.apache.kafka.connect.runtime.WorkerSourceTask.poll()`.

Also do not forget then when eithr stop or start genertaes the eception it might be non-retryeable. As in case of spike the first excepttion can be Borken Pipe and the attempt to re-establish connection will be for example SocketTimeoutException.

Reply:
I had been looking at this: https://cwiki.apache.org/confluence/display/KAFKA/KIP-298%3A+Error+Handling+in+Connect

It discusses two options:

* errors.retry.timeout
* errors.retry.delay.max.ms

But indeed it seems they're not applied to retrying the polling loop, but only for conversion and sending errors :(

So it seems the retry "feature" of KC is pretty much useless four our purposes.

Reply:
Ok, backoff implemented internally now via `ElapsedTimeStrategy` (still need to make the values configurable).